### ISSUE TITLE
Error when deploying an Approval Process with a 'starts with' criteria

### ISSUE THREAD

**Issue by RenaudAM** [forcedotcom/cli#2825]
> **Note** 
> Before you submit your issue, make sure that:
> - You're using the latest version of Salesforce CLI.
> - You've searched both open and closed issues for related posts.
> - You've used the `doctor` command to diagnose common issues. 
> - You understand that GitHub Issues don't adhere to any agreement or SLA.
>   - If you require immediate assistance, use official channels such as Salesforce Customer Support.

### Summary
Error when deploying an Approval Process with a 'starts with' criteria

### Steps To Reproduce

Deploying metadata using sf project deploy start command containing an approval process with a Step Criteria with a starts with condition.

### Expected result
Metadata deployed

### Actual result
In approval step XXX, the criteriaItems entry with object field XXX has an invalid value of 'starts with' for field 'operation'.

### System Information
Bash
$ sf version --verbose --json
{
  "architecture": "win32-x64",
  "cliVersion": "@salesforce/cli/2.36.8",
  "nodeVersion": "node-v20.12.1",
  "osVersion": "Windows_NT 10.0.19044",
  "rootPath": "C:\\Users\\A0468184\\AppData\\Roaming\\nvm\\v20.12.1\\node_modules\\@salesforce\\cli",
  "shell": "C:\\Program Files\\Git\\usr\\bin\\bash.exe",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 3.0.13 (core)",
    "@oclif/plugin-commands 3.2.2 (core)",
    "@oclif/plugin-help 6.0.20 (core)",
    "@oclif/plugin-not-found 3.1.1 (core)",
    "@oclif/plugin-plugins 5.0.5 (core)",
    "@oclif/plugin-search 1.0.20 (core)",
    "@oclif/plugin-update 4.2.3 (core)",
    "@oclif/plugin-version 2.0.16 (core)",
    "@oclif/plugin-warn-if-update-available 3.0.15 (core)",
    "@oclif/plugin-which 3.1.7 (core)",
    "@salesforce/cli 2.36.8 (core)",
    "apex 3.1.3 (core)",
    "auth 3.5.1 (core)",
    "data 3.2.4 (core)",
    "deploy-retrieve 3.5.3 (core)",
    "info 3.1.2 (core)",
    "limits 3.2.1 (core)",
    "marketplace 1.1.0 (core)",
    "org 3.6.3 (core)",
    "packaging 2.3.0 (core)",
    "schema 3.2.0 (core)",
    "settings 2.1.2 (core)",
    "sobject 1.2.1 (core)",
    "source 3.2.2 (core)",
    "telemetry 3.1.17 (core)",
    "templates 56.1.0 (core)",
    "trust 3.5.1 (core)",
    "user 3.4.3 (core)"
  ]
}

**Comment by WillieRuemmele** [comment]
Are you doing anything between the retrieve and deploy? After some googling, it seems like similar errors are thrown when something else references that [field](https://salesforce.stackexchange.com/a/87937)

**Comment by RenaudAM** [comment]
Hi @WillieRuemmele 
No this file never change since 2 years. I try to retrieve and deploy just after, no change after the retrieve and the deploy failed.

**Comment by WillieRuemmele** [comment]
@RenaudAM - did this just start happening for you after a CLI upgrade? if it's an api version difference you can use `--api-version` and specify a previous one... I'm trying to understand what change caused this to start happening

**Comment by RenaudAM** [comment]
Hi @WillieRuemmele 
Very difficult to understand. I tried different version of CLI ... no change, different api version ... no change. 
I tried in other Sandboxes, in some of them it works, in some others not.
I also tried to change startsWith by equals, deploy, change back to starts with in Salesforce, retrieve and deploy again ... but it failed.
For the moment I deploy with this file excluded because I have to deliver my release. I will try to understand more at the end of the week.
